bluemira/equilibria/optimisation/constraint_funcs.py

```diff
@@ -240,9 +240,23 @@ def df_constraint(self, vector: npt.NDArray[np.float64]) -> npt.NDArray[np.float
         Bz_a = self.az_mat @ currents
         B = np.hypot(Bx_a + self.bxp_vec, Bz_a + self.bzp_vec)
 
-        Bx = Bx_a * (Bx_a * currents + self.bxp_vec)
-        Bz = Bz_a * (Bz_a * currents + self.bzp_vec)
-        return np.round((Bx + Bz) / (B * self.scale**2), self._round_dp)
+        if len(B) == 1:
+            return np.round(
+                (self.scale / B)
+                * (
+                    self.ax_mat * (Bx_a + self.bxp_vec)
+                    + self.az_mat * (Bz_a + self.bzp_vec)
+                ),
+                self._round_dp,
+            )
+        return np.round(
+            (self.scale / B)[:, np.newaxis]
+            * (
+                self.ax_mat * (Bx_a + self.bxp_vec)[:, np.newaxis]
+                + self.az_mat * (Bz_a + self.bzp_vec)[:, np.newaxis]
+            ),
+            self._round_dp,
+        )
```

Comment:
Does this need to happen? I thought numpy would deal with the len(B)==1 implicitly?

Reply:
We get a mismatch of array sizes without it, probably has not come up before as we would normally have more than one field constraint.

Reply:
Then at least we're not repeating code
```suggestion
        res = np.round(
            (self.scale / B)[:, np.newaxis]
            * (
                self.ax_mat * (Bx_a + self.bxp_vec)[:, np.newaxis]
                + self.az_mat * (Bz_a + self.bzp_vec)[:, np.newaxis]
            ),
            self._round_dp,
        )

        if len(B) == 1:
            return np.squeeze(res)
        return res
```